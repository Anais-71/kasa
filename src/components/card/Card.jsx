import React, { useState, useEffect } from 'react';
import { fetchData } from '../../utils/fetch';

//CSS
import './card.css'

function Card() {
    const [data, setData] = useState([]);

    useEffect(() => {
        const getData = async () => {
            const dataJson = await fetchData();
            setData(dataJson);
        };
        getData();
    }, []);

    return (
        <div className='lodges'>
            {data.map((item, index) => (
                <a href={'/logement/' + item.id} className='card' key={index}>
                    <img src={item.cover} alt="logement" className='card__cover'/>
                    <div className='card__overlay'></div>
                    <p className='card__title' data-testid='title'> {item.title} </p>
                </a>
            ))}
        </div>
    )
}

export default Card;