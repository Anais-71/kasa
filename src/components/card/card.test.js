import { render, fireEvent } from '@testing-library/react';
import Card from './Card';

describe('Given I am on the home page', () => {
  const title = 'Test title';
  const cover = ['https://s3-eu-west-1.amazonaws.com/course.oc-static.com/projects/front-end-kasa-project/accommodation-20-1.jpg'];

  test('When the card component is being displayed, Then the title should be present', () => {
    const { getByText } = render(<Card title={title} />);
    expect(getByText(title)).toBeInTheDocument();
  });  

  test('When the card component is being displayed, Then the cover image should be present', () => {
    const { getByText } = render(<Card cover={cover} />);
    expect(getByText(cover)).toBeInTheDocument();
  });

  test('The dropdown page matches the snapshot', () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });
});