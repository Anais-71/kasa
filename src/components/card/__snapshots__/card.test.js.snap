// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given I am on the home page The dropdown page matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="lodges"
  />
</DocumentFragment>
`;
